core/src/test/java/org/elasticsearch/discovery/DiscoveryWithServiceDisruptionsIT.java

```diff
@@ -512,11 +523,15 @@ public void run() {
                     assertThat(semaphore.availablePermits(), equalTo(0));
                     semaphore.release(docsPerIndexer);
                 }
-                assertTrue(countDownLatchRef.get().await(60000 + disruptionScheme.expectedTimeToHeal().millis() * (docsPerIndexer * indexers.size()), TimeUnit.MILLISECONDS));
+                logger.info("waiting for indexing requests to complete");
+                assertTrue(countDownLatchRef.get().await(docsPerIndexer * 1000 + 2000, TimeUnit.MILLISECONDS));
```

Comment:
hmm... I'm not sure about this. It means we are not indexing when the partition is healed... not sure why I changed it. I think it was to make sure that we have some indexing during disruption but looking at it now I think we should split into rounds - indexing while disrupting , stop disrupt and index some more, then wait for the cluster to heal. WDYT?